spec/controllers/v0/user_action_events_controller_spec.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V0::UserActionEventsController, type: :controller do
+  include RequestHelper
+
+  context 'when not logged in' do
+    it 'returns unauthorized' do
+      get :index
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  describe 'GET #index' do
```

Comment:
The invocation of the `index` call for  each test can be standardized into a `subject` declared at the top of the method's test code.

```ruby
subject { get(:index, params: index_params) }
let(:index_params) { { start_date:, end_date:, page:, per_page: } }
let(:start_date) { nil }
let(:end_date) { nil }
let(:page) { nil }
let(:per_page) { nil }
```

Then for each individual test you can choose to replace the default behavior with what you specifically need.